.github/workflows/build-and-test.yml

```diff
@@ -53,7 +55,7 @@ jobs:
   build-and-test:
     runs-on: ${{ matrix.os }}
     needs: check-for-modification
-    if: ${{ needs.check-for-modification.outputs.core-modified == 'true' }}
+    if: ${{ (!github.event.pull_request || github.event.pull_request.draft == false) && needs.check-for-modification.outputs.core-modified == 'true' }}
```

Comment:
Can we make the check its own job and have all other jobs in this workflow specify it in `needs`? That would avoid the duplicate if-conditions if nothing much else.